### PR TITLE
Added some API docs

### PR DESCRIPTION
**Pull request by BrettRToomey**

### CODE REVIEW

File: Sources/AdminPanelProvider/Middlewares/PanelConfigMiddleware.swift [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+/// AdminPanel's configuration
 public struct PanelConfig {
+    /// Name to use for emails and the site's logo
     public let panelName: String
+    /// Base URL of the admin panel
     public let baseUrl: String
+    /// AdminLTE skin colour
     public let skin: Skin
 
+    /// Is it?
```
Comment (steffendsommer) [inline Sources/AdminPanelProvider/Middlewares/PanelConfigMiddleware.swift]: I think we should explain this - as with Storage further down.

**Review by steffendsommer — APPROVED** [review]

**Comment by steffendsommer** [comment]
Please add the login to the SSO package so we it doesn't get lost for the next dev pulling in the admin panel provider.

**Comment by steffendsommer** [comment]
This LGTM. Do you mind creating the PR for nodes-sso so we don't forget the custom login page?

**Comment by martinlasek** [comment]
related to issue #13, #15

File: README.md [hunk 2]
```diff
@@ -103,22 +103,34 @@ The user that will be created using the seeder will have the following credentia
...
+| `baseUrl`          | `http://mytechblog.com`                                                                | No       | This will be used to generate urls for the admin panel (e.g. when resetting a password).                                                                                                                                                      |
+| `skin`             | `green-light`                                                                          | No       | The skin to use for the admin panel. The options will correspond to the [available skins](https://adminlte.io/themes/AdminLTE/documentation/index.html#layout) supported by AdminLTE. Please omit the `skin-` prefix when specifying the skin.|
+| `email`            | `{"fromName": "Admin Panel", "fromAddress": "admin@panel.com"}`                       `| No       | This will be used to configure the AdminPanel's mailer                                                                                                                                                                                        |
```
Comment (steffendsommer) [inline README.md]: Missing full stop ".. mailer."
Will it work without setting this?

File: Sources/AdminPanelProvider/Provider.swift [hunk 3]
```diff
@@ -24,29 +24,38 @@ public final class Provider: Vapor.Provider {
...
+            if let userSkinConfig = config["skin"]?.string {
+                skin = PanelConfig.Skin(rawValue: userSkinConfig) ?? skin
+            }
+
+            if let url = config["baseUrl"]?.string {
+                baseUrl = url
+            }
         } else {
-            print("WARNING: couldn't find `mailgun.json`. Email features will be disabled.")
+            print("WARNING: couldn't find `\(fileName).json`. Email features will be disabled.")
```
Comment (steffendsommer) [inline Sources/AdminPanelProvider/Provider.swift]: I think we need to consider the case where the file is present, but the email stuff is not configured.

**Review by steffendsommer — APPROVED** [review]
Some minor things, otherwise LGTM. Remember to update and release Nodes-SSO with the index page and make sure this new release uses the new release of Nodes-SSO.

**Pull request merged**